framer/Color.coffee

```diff
@@ -1,3 +1,468 @@
-class Color extends BaseClass
+{BaseClass} = require "./BaseClass"
 
-	constructor: ->
\ No newline at end of file
+class exports.Color extends BaseClass
+	constructor: (@color) ->
+
+		color = @color
+
+		# If input already is a Color object return input
+		if (color instanceof Color) then return color
+
+		# Convert input to RGB
+		rgb = inputToRGB(color)
+
+		@_r = rgb.r
+		@_g = rgb.g
+		@_b = rgb.b
+		@_a = rgb.a
+		@_roundA = Math.round(100*@_a) / 100
+
+	@define "r",
+		get: -> @_r
+		set: (value) -> throw "r is readonly"
```

Comment:
I'm pretty sure these get thrown automatically if you don't have a setter. Check BaseClass. You should use those. Also, officially, you need to throw a `new Error`, I think.